src/main/java/core/basesyntax/service/operations/SupplyOperation.java

```diff
@@ -0,0 +1,19 @@
+package core.basesyntax.service.operations;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.exceptions.InvalidFruitException;
+
+public class SupplyOperation implements OperationHandler {
+    @Override
+    public void apply(FruitTransaction transaction) {
+        if (Storage.containsKey(transaction.getFruit())) {
+            int newValue = Storage.getQuantity(transaction.getFruit())
+                    + transaction.getQuantity();

```

Comment:
There is no check for negative values. Consider adding a condition to check if the transaction quantity is not negative.

Reply:
ok, done